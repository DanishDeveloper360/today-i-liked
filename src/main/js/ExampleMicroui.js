import React, { useEffect, useRef } from 'react';
function ExampleMicroui() {
  const autocompleteRef = useRef();

  let { firstName } = window.scopus.platform.user.identification.getIdentity();

  const countries = [
    { id: 1, label: 'Afghanistan' },
    { id: 2, label: 'Albania' },
    { id: 3, label: 'Algeria' },
    { id: 4, label: 'Angola' },
    { id: 5, label: 'Armenia' },
    { id: 6, label: 'Argentina' },
    { id: 7, label: 'Congo' },
    { id: 8, label: 'Croatia' },
    { id: 9, label: 'France' },
    { id: 10, label: 'Netherlands' },
  ];

  const ICONS = [
    'apple',
    '3d-radiology',
    'cooking-tools',
    'couple-of-people',
    'delivery-van',
    'descend',
  ];

  useEffect(() => {
    autocompleteRef.current.onExactSearch = (query) => {
      alert(`Exact search done with query: ${query}`);
    };
    autocompleteRef.current.onUserTyping = (query) => {
      autocompleteRef.current.suggestions = countries.filter(
        (country) => country.label.indexOf(query) !== -1
      );
    };
    autocompleteRef.current.onSuggestionSelect = (suggestion) => {
      alert(`Suggestion ${suggestion.label} selected`);
    };
    autocompleteRef.current.config = {
      queryTresholdLength: 2,
      debounceTimeout: 500,
      placeholder: 'Filter a country',
      showClearIcon: true,
      showSearchIcon: true,
      small: true,
    };
  }, []);

  return (
    <div id="ScopusExampleMicroui">
      <div className="container sc-ExampleMicroui--flex">
        <section className="row">
          <div className="col-24">
            <h1>Scopus ExampleMicroui Example</h1>
            <p>Hello, {firstName}! Welcome to Scopus.</p>
            <p className="text-meta--small">
              This is a full column based on a 24 column grid layout.
            </p>
            <sc-autocomplete
              ref={autocompleteRef}
              id="autocomplete-example"
            ></sc-autocomplete>
          </div>
        </section>
        <section className="margin-size-24-t">
          <div className="row">
            <h2 className="col">Icons</h2>
          </div>
          <div className="row">
            {ICONS.map((icon) => (
              <div className="col-12 col-md-4" key={icon}>
                <div className="box">
                  <sc-icon
                    icon-name={icon}
                    onClick={() => alert(`clicked ${icon}`)}
                  ></sc-icon>
                </div>
              </div>
            ))}
          </div>
        </section>
      </div>
    </div>
  );
}

ExampleMicroui.propTypes = {};

export default ExampleMicroui;
