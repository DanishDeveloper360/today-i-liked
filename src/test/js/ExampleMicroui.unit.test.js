import React from 'react';
import { mount } from 'enzyme';

import ExampleMicroui from '../../main/js/ExampleMicroui';

describe('ExampleMicroui', () => {
  let testExampleMicroui;

  beforeEach(async () => {
    global.scopus = {
      platform: {
        configuration: {
          getComponentConfiguration() {
            return {
              FOO_ENABLED: true,
              BAR_ENABLED: false,
            };
          },
        },
        user: {
          identification: {
            getIdentity: jest.fn(() => ({
              firstName: 'example-microui-username',
            })),
          },
        },
      },
    };
    global.alert = jest.fn();
    testExampleMicroui = mount(<ExampleMicroui />);
  });

  describe('Snapshot test', () => {
    it('should render a snapshot', () => {
      expect(testExampleMicroui.html()).toMatchSnapshot();
    });
  });
});
