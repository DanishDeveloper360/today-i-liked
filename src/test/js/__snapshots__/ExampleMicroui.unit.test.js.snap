// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleMicroui Snapshot test should render a snapshot 1`] = `"<div id=\\"ScopusExampleMicroui\\"><div class=\\"container sc-ExampleMicroui--flex\\"><section class=\\"row\\"><div class=\\"col-24\\"><h1>Scopus ExampleMicroui Example</h1><p>Hello, example-microui-username! Welcome to Scopus.</p><p class=\\"text-meta--small\\">This is a full column based on a 24 column grid layout.</p><sc-autocomplete id=\\"autocomplete-example\\"></sc-autocomplete></div></section><section class=\\"margin-size-24-t\\"><div class=\\"row\\"><h2 class=\\"col\\">Icons</h2></div><div class=\\"row\\"><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"apple\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 92 128\\" height=\\"24\\" width=\\"18\\" aria-hidden=\\"true\\"><path d=\\"M56.93 33.02c-.21-2.12-.6-4.8-1.23-7.56 4.03.74 9.29.97 14.05-1.14 9.39-4.17 13.27-15.86 13.27-15.86s-11.32-4.96-20.7-.8c-4.27 1.9-7.38 5.34-9.53 8.54-1.21-3-2.72-6.14-4.68-9.3l-8.49 5.26c3.81 6.16 6.23 13.1 7.18 20.28.67 5.12.64 10.44-.12 15.54l7.65-3.72c.11-.04 11.59-5.5 19.46.12 7.07 5.04 9.25 19.86 5.31 35.98-3.08 12.62-9.45 22.73-14.81 23.5-3.8.55-5.16-.06-7.87-1.72-2.13-1.3-4.78-2.94-8.8-3.82-.99-.22-1.93-.3-2.85-.3-4.05 0-7.39 1.88-10.17 3.46-3.95 2.24-6.37 3.2-10.2 1.78-5.47-2.04-12.48-17.18-13.32-33.64-.34-6.7-.01-22.81 10.45-27.96 6.44-3.17 12.89-1.14 17.08 1.06.21-3.84.11-8.22-.4-11-5.86-2.02-13.35-2.83-21.1.98C5.99 38.17.16 51.8 1.1 70.12c.89 17.38 8.14 38.17 19.82 42.5 8.37 3.11 14.29-.24 18.6-2.68 2.77-1.56 4.41-2.43 5.96-2.08 2.37.53 3.92 1.46 5.72 2.56 2.64 1.62 5.83 3.58 11.08 3.58 1.05 0 2.19-.09 3.42-.26 11.87-1.69 19.72-17.12 23.11-31 3.83-15.68 3.71-37.1-9.21-46.34-7.68-5.5-16.62-4.84-22.67-3.38z\\"></path></svg></sc-icon></div></div><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"3d-radiology\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 126 128\\" height=\\"24\\" width=\\"24\\" aria-hidden=\\"true\\"><path d=\\"M48 68.5v18.32c0 5.78-2.04 10.74-6.08 14.7-6.48 6.4-15.98 8.44-19.26 8.4-7.18-.1-10.66-4.46-10.66-13.34 0-20.48 8.76-51.82 20.08-61.68 4.42-3.86 6.6-2.86 7.32-2.52 2.1.96 3.94 3.36 5.4 6.44 2.64-3.02 4.4-6.74 4.98-10.74-1.74-2.08-3.8-3.7-6.22-4.8-4-1.82-10.38-2.58-18.04 4.08C10.38 40.56 2 76.62 2 96.58c0 14.44 7.68 23.16 20.52 23.34h.24c5.8 0 17.82-3.02 26.2-11.28C54.88 102.8 58 95.26 58 86.82V61.98c-2.72 2.12-3.8 2.7-10 6.52zm52.5-41.14c-7.66-6.68-14.04-5.9-18.04-4.08-2.44 1.1-4.48 2.74-6.22 4.82.58 4 2.34 7.72 4.98 10.74 1.32-2.8 3.92-6.82 6.96-6.82 1.2 0 3.06.56 5.76 2.88C105.24 44.74 114 76.1 114 96.58c0 8.86-3.48 13.24-10.66 13.34-3.34 0-12.78-2-19.26-8.4C80.04 97.56 78 92.6 78 86.82V68.5c-6.18-3.8-7.28-4.38-10-6.52v24.84c0 8.44 3.12 15.98 9.06 21.82 8.38 8.26 20.4 11.28 26.2 11.28h.24c12.82-.18 20.5-8.9 20.5-23.34 0-19.96-8.38-56.04-23.5-69.22zM76.4 58.12l14.22 8.76 5.24-8.52-14.24-8.76C73.22 44.42 68 35.06 68 25.18V6H58v19.18c0 9.88-5.22 19.24-13.64 24.42l-14.24 8.76 5.24 8.52 14.22-8.76c5.64-3.48 10.22-8.34 13.4-14 3.2 5.66 7.78 10.52 13.42 14z\\"></path></svg></sc-icon></div></div><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"cooking-tools\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 112 128\\" height=\\"24\\" width=\\"21\\" aria-hidden=\\"true\\"><path d=\\"M96 62h-9V6H77v56h-9c-8.29 0-15 6.72-15 15v30c0 8.28 6.71 15 15 15h1v-10h-1c-2.76 0-5-2.24-5-5V77c0-2.76 2.24-5 5-5h28c2.76 0 5 2.24 5 5v30c0 2.76-2.24 5-5 5h-9V80H77v42h19c8.29 0 15-6.72 15-15V77c0-8.28-6.71-15-15-15M37 49c0 7.16-5.83 13-13 13s-13-5.84-13-13V6H1v43c0 10.96 7.72 20.14 18 22.44V122h10V71.44C39.28 69.14 47 59.96 47 49V6H37v43\\"></path></svg></sc-icon></div></div><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"couple-of-people\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 118 128\\" height=\\"24\\" width=\\"23\\" aria-hidden=\\"true\\"><path d=\\"M37 31.52c4.27 0 11.44 1.38 11.44 10.67 0 8.28-4.81 14.29-11.44 14.29s-11.44-6.01-11.44-14.29c0-9.29 7.17-10.67 11.44-10.67zM37 66c11.97 0 21-10.21 21-23.74C58 30.14 49.56 22 37 22s-21 8.14-21 20.26C16 55.79 25.03 66 37 66m47-24.44c2.44 0 10.4.7 10.4 9.73 0 7.62-4.37 13.15-10.4 13.15s-10.4-5.53-10.4-13.15c0-9.03 7.96-9.73 10.4-9.73zM84 74c11.4 0 20-9.74 20-22.64C104 39.96 95.78 32 84 32s-20 7.96-20 19.36C64 64.26 72.6 74 84 74m-46.97-2C7.6 72 1.32 87.2.43 96.34L.2 108h9.5l.29-10.96c.99-10.12 10.62-15.24 27.04-15.24 16.43 0 26.06 5.13 27.04 15.16l.26 11.04h9.64l-.34-11.74C72.75 87.2 66.46 72 37.03 72m80.52 30.94c-1.4-14.57-13.47-22.92-33.45-22.92-3.29 0-6.35.24-9.21.69 1.92 2.78 3.35 5.89 4.27 9.31 1.55-.13 3.19-.2 4.94-.2 14.83 0 23.01 4.67 23.89 13.81l.36 4.37h9.62l-.42-5.06\\"></path></svg></sc-icon></div></div><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"delivery-van\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 128 128\\" height=\\"24\\" width=\\"24\\" aria-hidden=\\"true\\"><path d=\\"M36 40H0v10h36V40zM0 22v10h94.39c2.2 0 4.17 1.47 4.79 3.58L103.46 50H86V40H76v12.98c0 3.86 2.99 7.02 6.85 7.02h26.6c7.26 5.84 8.56 6.35 8.56 14v12h-4.91c-4.91-16-27.57-16-32.49 0H55.1c-4.91-16-27.57-16-32.5 0H18V68h8V58H0v10h8v28h14.6c4.91 16 27.57 16 32.5 0h25.51c4.91 16 27.57 16 32.5 0h14.75c0-36 1.85-33.05-13.67-44.98l-5.42-18.28C106.9 26.41 100.98 22 94.39 22H0zm39.15 61.89c1.54.05 3.14.67 4.59 2.09 6.63 6.47-3.15 16.46-9.77 10-5.18-5.06-.34-12.26 5.18-12.09zm57.71.09c6.19 0 9.4 7.59 4.88 12-4.45 4.35-11.88 1.12-11.88-5 0-3.84 3.15-7 7-7z\\"></path></svg></sc-icon></div></div><div class=\\"col-12 col-md-4\\"><div class=\\"box\\"><sc-icon icon-name=\\"descend\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 108 128\\" height=\\"24\\" width=\\"21\\" aria-hidden=\\"true\\"><path d=\\"M54 92.242L.315 36h107.37zM23.685 46L54 77.758 84.315 46z\\"></path></svg></sc-icon></div></div></div></section></div></div>"`;
